x/bsonx/bsoncore/document.go

```diff
@@ -291,6 +291,91 @@ func (d Document) String() string {
 	return buf.String()
 }
 
+// StringN stringifies a document upto N bytes
+func (d Document) StringN(n int) string {
+	if len(d) < 5 || n <= 0 {
+		return ""
+	}
+
+	// String outputted is short by 2
+	n += 2
+
+	var buf strings.Builder
+	buf.Grow(n) //Preallocate buffer size
```

Comment:
This could cause very large and unnecessary allocations if someone passes in a large value of `n`. For example, if a user has a BSON document with 1 field, but calls `StringN(1_000_000)`, this will allocate 1Mb of space and then do nothing with it. There's not currently a simple way to predict the size of the resulting Extended JSON document based on the BSON document, so we should only pre-allocate if `n` is small.

E.g.
```go
if n <= 1024 {
	buf.Grow(n)
}
```